spec/Section 2 -- Source Schema.md

```diff
@@ -2,59 +2,120 @@
 
 ## Directives
 
-### @entityResolver
+### @lookup
 
 ```graphql
-directive @entityResolver on FIELD_DEFINITION
+directive @lookup on FIELD_DEFINITION
 ```
 
-Entity resolvers are fields on the query root type of a subgraph that can
-resolve an entity by a stable key. The stable key is defined by the arguments of
-the field.
+The `@lookup` directive is used within a _source schema_ to specify object
+fields that can be used by the _distributed GraphQL executor_ to resolve an
+entity by a stable key.
+
+The stable key is defined by the arguments of the field. Only fields that are
+annotated with the `@lookup` directive will be recognized as lookup field.
+
+Source schemas can provide multiple lookup fields for the same entity with
+different keys.
+
+In this example the source schema specifies that the `Person` entity can be
+resolved with the `personById` field or the `personByName` field on the `Query`
+type. Both fields can resolve the same entity but do so with different keys.
 
 ```graphql example
 extend type Query {
-  version: Int # NOT an entity resolver.
-  personById(id: ID!): Person @entityResolver
+  version: Int # NOT a lookup field.
+  personById(id: ID!): Person @lookup
+  personByName(name: String!): Person @lookup
 }
 
-extend type Person {
-  id: ID! # matches the argument of personById
+extend type Person @key(fields "id")  @key(fields "name") {
```

Comment:
should we drop all of those `extend`s? this was required in Federation v1 but since then we moved away from it and I don't think it should be necessary to define those types as extensions

Reply:
➕

Reply:
I mainly added the extend as I am not having the complete type here.

Reply:
I will update the PR to remove confusion.